src/garage/tf/embeddings/gaussian_mlp_encoder.py

```diff
@@ -118,11 +137,15 @@ def __init__(self,
     def build(self, embedding_input, name=None):
         """Build encoder.
 
+        After build, self.distribution is a Gaussian distribution conitioned
```

Comment:
Would it still be possible to build this encoder with a different input, and retrieve the resulting outputs, e.g. distribution? It seems like enforcing build to assign an attribute is not what we want. Instead we may build in constructor and assign these attributes, and build should return the outputs.

Reply:
I agree. `build()` better has no side-effect.